packages/core/src/App/Containers/Modals/trading-assessment-existing-user.jsx

```diff
@@ -0,0 +1,174 @@
+
+        return () => setIsTradingAssessmentForExistingUserEnabled(false);
+    }, []);
+
+    const handleSubmit = async values => {
+        if (values.risk_tolerance === 'No') {
+            setShouldShowTradeAssessmentForm(false);
+            setShouldShowRiskToleranceWarningModal(true);
+        } else {
+            const form_payload = {
+                trading_experience_regulated: { ...values },
+            };
+            const response = await setFinancialAndTradingAssessment(form_payload);
+            const { trading_score } = response.set_financial_assessment ?? {};
+            await updateAccountStatus();
+            setShouldShowTradeAssessmentForm(false);
+            if (trading_score === 0) {
+                setShouldShowAppropriatenessTestWarningModal(true);
+            } else {
+                setShouldShowAssessmentCompleteModal(true);
+            }
+        }
+    };
+
+    const handleAcceptAppropriatenessTestWarning = () => {
+        setShouldShowAppropriatenessTestWarningModal(false);
+        setShouldShowAssessmentCompleteModal(true);
+    };
+
+    const handleAcceptRisk = () => {
+        setFormValue(prev_state => ({ ...prev_state, risk_tolerance: 'Yes' }));
+        setShouldMoveToNext(true);
+        setShouldShowRiskToleranceWarningModal(false);
+        setShouldShowTradeAssessmentForm(true);
+    };
+
+    if (should_show_risk_tolerance_warning_modal) {
+        return (
+            <RiskToleranceWarningModal
+                show_risk_modal={should_show_risk_tolerance_warning_modal}
+                title={localize('Risk Tolerance Warning')}
+                button_text={localize('Yes, I understand the risk.')}
+                onClick={handleAcceptRisk}
+                body_content={
+                    <Localize
+                        i18n_default_text='CFDs and other financial instruments come with a high risk of losing money rapidly due to leverage. You should consider whether you understand how CFDs and other financial instruments work and whether you can afford to take the high risk of losing your money. <0/><0/> To continue, you must confirm that you understand your capital is at risk'
+                        components={[<br key={0} />]}
+                    />
+                }
+            />
+        );
+    } else if (should_show_appropriateness_test_warning_modal) {
+        return (
+            <TestWarningModal
+                show_risk_modal={should_show_appropriateness_test_warning_modal}
+                body_content={
+                    <Text as='p' size='xs'>
+                        <Localize
+                            i18n_default_text='In providing our services to you, we are required to ask you for some information to assess if a given product or service is appropriate for you and whether you have the experience and knowledge to understand the risks involved.<0/><1/>'
+                            components={[<br key={0} />, <br key={1} />]}
```

Comment:
Why is this declaring two identical elements with different keys?

Reply:
Resolved 🫡